src/ui/components/results/paginationcomponent.js

```diff
@@ -79,10 +90,11 @@ export default class PaginationComponent extends Component {
 
   shouldShowControls (results, limit) {
     const hasResults = results.searchState === 'search-complete' && results.resultsCount > limit;
-    const noResultsConfig = this.core.globalStorage.getState(StorageKeys.NO_RESULTS_CONFIG) || {};
-    const showControls = hasResults &&
-      (results.resultsContext === ResultsContext.NORMAL || noResultsConfig.displayAllResults);
-    return showControls;
+    const isNormalResults = results.resultsContext === ResultsContext.NORMAL;
+    const isVisibleForNoResults = 'visible' in this._noResults
```

Comment:
Does `this._noResults` fall back to an empty object? If so, could this be `this._noResults.visible || this._noResults. displayAllResults`?

Reply:
If they set noResults.visible to be false I didn't want it to be true if displayAllResults happened to be true as well. I doubt anybody would ever set both but it feels a little more accurate to me this way